### PR TITLE
Zendoo support

### PR DESCRIPTION
**Pull request by otoumas**
This PR has all the changes needed to add Zendoo Support to the library

### CODE REVIEW

**Review by carlosper88 — APPROVED** [review]

File: lib/transaction/transaction.js [hunk 1]
```diff
@@ -353,9 +458,34 @@ Transaction.prototype.toObject = Transaction.prototype.toJSON = function toObjec
...
+    obj.scProof =  this.scProof;
+    obj.vFieldElementCertificateField = this.vFieldElementCertificateField;
+    obj.vBitVectorCertificateField = this.vBitVectorCertificateField;
+    obj.ftScFeeBN = this.ftScFeeBN;
+    obj.ftScFee = this.ftScFee;
+    obj.mbtrScFeeBN = this.mbtrScFeeBN;
+    obj.mbtrScFee = this.mbtrScFee;
+  }
+
+  if (this.version >= 2 || this.version === Transaction.SHIELDED_TX) {
```
Comment (paolocappelletti) [inline lib/transaction/transaction.js:488]: replace 2 with Transaction.SHIELDED_TX_OLD

File: lib/address.js [hunk 2]
```diff
@@ -116,6 +115,10 @@ Address.prototype._classifyArguments = function(data, network, type) {
 Address.PayToPublicKeyHash = 'pubkeyhash';
 /** @static */
 Address.PayToScriptHash = 'scripthash';
+/** @static */
+Address.PayToPublicKeyHashReplay = 'pubkeyhashreplay';
+/** @static */
+Address.ScriptHash = 'scripthash';
```
Comment (paolocappelletti) [inline lib/address.js:121]: looking at the code the Address.ScriptHash  appears to me to be the pay to script hash with replay protection, so to be coherent with the previous ones I would rename it to:
Address.PayToScriptHashReplay= 'scripthashreplay';
please check all the  "if"  using it and change accordingly
Also why we have used the same constant value of standard PayToScriptHash? seems better to discriminate it, and if in some places we need not handle it in different ways for backward compatibility add an if clause

File: lib/address.js [hunk 3]
```diff
@@ -66,12 +66,11 @@ function Address(data, network, type) {
     throw new TypeError('Second argument must be "livenet" or "testnet".');
   }
 
-  if (type && (type !== Address.PayToPublicKeyHash && type !== Address.PayToScriptHash)) {
+  if (type && (type !== Address.PayToPublicKeyHash && type !== Address.PayToScriptHash && type !== Address.PayToPublicKeyHashReplay)) {
```
Comment (paolocappelletti) [inline lib/address.js:69]: add PayToScriptHashReplay (see below comment) and extend the error message

File: lib/address.js (hunk 2 above)
Comment (otoumas) [reply lib/address.js:121]: I have created a ticket to do these changes after zendoo launch

**Review by cronicc — APPROVED** [review]

**Pull request merged**